### PR TITLE
Remove broken missing products check to avoid accidentally using it. 

### PR DESCRIPTION
**Pull request by EmileSonneveld**
Replaces this PR: https://github.com/Open-EO/openeo-geopyspark-driver/pull/584

### CODE REVIEW

File: openeogeotrellis/layercatalog.py [hunk 1]
```diff
@@ -1045,8 +1045,9 @@ def check_missing_products(
 
         method = check_data.get("method")
         if method == "creo":
-            creo_catalog = CreoCatalogClient(**check_data["creo_catalog"])
-            missing = [p.getProductId() for p in creo_catalog.query_offline(**query_kwargs)]
```
Comment (soxofaan) [inline openeogeotrellis/layercatalog.py:1049]: maybe also put a TODO comment in the `query_offline` explaining why the implementation needs work

**Comment by EmileSonneveld** [comment]
@soxofaan Can this check be removed then? https://github.com/Open-EO/openeo-geopyspark-driver/commit/c63728c3cad9fb71a63bb031d6561ad71880e3c1#diff-e48ae7baaf533de3301112527309f195fef859e933e1e17a1955b3bc833af33aR1513

File: openeogeotrellis/layercatalog.py (hunk 1 above)
Comment (EmileSonneveld) [reply openeogeotrellis/layercatalog.py:1049]: Added a comment

**Comment by soxofaan** [comment]
indeed, that if from c63728c3cad9fb71a6  can also be removed with this PR

**Comment by EmileSonneveld** [comment]
So, the broken missing products check is removed, the test for that code too, and `extra_validation` could be re-enabled. @soxofaan , this is good to merge, right?

**Pull request merged**